src/inject.ts

```diff
@@ -0,0 +1,620 @@
+} from "./utils.js";
+
+type AllMatches = {
+	searchee: SearcheeWithLabel;
+	decision: DecisionAnyMatch;
+}[];
+type FullMatches = {
+	searchee: SearcheeWithLabel;
+	decision: Decision.MATCH | Decision.MATCH_SIZE_ONLY;
+}[];
+type PartialMatches = {
+	searchee: SearcheeWithLabel;
+	decision: Decision.MATCH_PARTIAL;
+}[];
+
+type InjectSummary = {
+	TOTAL: number;
+	INJECTED: number;
+	FULL_MATCHES: number;
+	PARTIAL_MATCHES: number;
+	BLOCKED: number;
+	ALREADY_EXISTS: number;
+	INCOMPLETE_CANDIDATES: number;
+	INCOMPLETE_SEARCHEES: number;
+	FAILED: number;
+	UNMATCHED: number;
+	FOUND_BAD_FORMAT: boolean;
+	FLAT_LINKING: boolean;
+};
+
+function getTorrentFilePathLog(torrentFilePath: string): string {
+	return chalk.bold.magenta(
+		torrentFilePath.replace(/\[([a-z0-9]{40})].torrent$/i, (match, hash) =>
+			match.replace(hash, sanitizeInfoHash(hash)),
+		),
+	);
+}
+
+function deleteTorrentFileIfElligible(
+	torrentFilePath: string,
+	options: { isComplete: boolean; cleanUpOldTorrents: boolean },
+): void {
+	const filePathLog = getTorrentFilePathLog(torrentFilePath);
+	try {
+		if (options?.isComplete) {
+			logger.verbose({
+				label: Label.INJECT,
+				message: `Deleting ${filePathLog} as it's in client and complete`,
+			});
+			fs.unlinkSync(torrentFilePath);
+			return;
+		}
+		if (!options.cleanUpOldTorrents) return;
+		if (fs.statSync(torrentFilePath).mtimeMs < Date.now() - ms("1 week")) {
+			logger.warn({
+				label: Label.INJECT,
+				message: `Deleting ${filePathLog} as it has failed to inject for too long`,
+			});
+			fs.unlinkSync(torrentFilePath);
+		}
+	} catch (e) {
+		logger.error({
+			label: Label.INJECT,
+			message: `Failed to delete ${filePathLog}`,
+		});
+		logger.debug(e);
+	}
+}
+
+async function injectDecideStage(
+	meta: Metafile,
+	searchees: SearcheeWithLabel[],
+): Promise<{
+	fullMatches: FullMatches;
+	partialMatches: PartialMatches;
+	foundBlocked: boolean;
+}> {
+	const fullMatches: FullMatches = [];
+	const partialMatches: PartialMatches = [];
+	let foundBlocked = false;
+	for (const searchee of searchees) {
+		const { decision } = await assessCandidate(meta, searchee, []);
+		if (!isAnyMatchedDecision(decision)) {
+			if (decision === Decision.BLOCKED_RELEASE) {
+				foundBlocked = true;
+			}
+			continue;
+		}
+
+		// If name or file names are not similar consider it a false positive
+		if (
+			!areMediaTitlesSimilar(searchee.title, meta.title) &&
+			!areMediaTitlesSimilar(searchee.title, meta.name) &&
+			!areMediaTitlesSimilar(searchee.name, meta.name) &&
+			!areMediaTitlesSimilar(searchee.name, meta.title) &&
+			!meta.files.some((metaFile) =>
+				searchee.files.some((searcheeFile) =>
+					areMediaTitlesSimilar(searcheeFile.name, metaFile.name),
+				),
+			)
+		) {
+			logger.warn({
+				label: Label.INJECT,
+				message: `Skipping potential false positive for ${getLogString(meta, chalk.bold.white)} from ${getLogString(searchee, chalk.bold.white)}`,
+			});
+			continue;
+		}
```

Comment:
What does this protect us from? MATCH_SIZE_ONLY matches that may be incorrect?

Reply:
Would it not be prudent to check `matchMode !== MatchMode.SAFE` prior to checking the titles if this is the case? Otherwise, it could do this unnecessarily?

Reply:
It protects against bad matches from risky or partial. Think of it similar to how torznab search or reverselookup requires titles to be similar.

This is largely unnecessary for inject though as every .torrent in outputDir already passed the above filters and more, but it could technically match with another unrelated torrent a user has.

I don't think it's necessary to check for matchMode as the first check passes in 100% of cases.

Reply:
shouldn't this code live in `decide` then?

Reply:
`areMediaTitlesSimilar` could go in there but I think it's fine in utils. And the use of it is specific and straightforward, I don't think a wrapper function is needed.

Reply:
what I'm saying is that this sounds like a decision rejection. Why wouldn't this make sense at the end of assessCandidateHelper?

Reply:
I don't think it adds any benefit to any other method. Meaning if this check fails, then it wouldn't have matched reverse lookup or been in the search results. This is just here to bring the safety closer to those methods, but it's not as strict as search and rss/announce.